src/Controls/src/Core/INativeBindingService.cs

```diff
@@ -1,9 +1,12 @@
 #nullable disable
+using System.Diagnostics.CodeAnalysis;
+
 namespace Microsoft.Maui.Controls.Xaml.Internals
 {
 
 	public interface INativeBindingService
 	{
+		[RequiresUnreferencedCode(TrimmerConstants.StringPathBindingWarning, Url = TrimmerConstants.ExpressionBasedBindingsDocsUrl)]
 		bool TrySetBinding(object target, string propertyName, BindingBase binding);
```

Comment:
Generally putting [RUC] on an interface is not great, but because it's inside `Microsoft.Maui.Controls.Xaml.Internals` seems ok.

Reply:
I was looking at all the `INativeBindingService` implementations, and they all do a check that would throw:
https://github.com/dotnet/maui/blob/b183cae2982f7c318f796d202f5c811d4da0edd8/src/Compatibility/Core/src/iOS/NativeBindingService.cs#L17
https://github.com/dotnet/maui/blob/b183cae2982f7c318f796d202f5c811d4da0edd8/src/Controls/src/Core/Hosting/Effects/AppHostBuilderExtensions.cs#L20

Unless you call:
https://github.com/dotnet/maui/blob/b183cae2982f7c318f796d202f5c811d4da0edd8/src/Compatibility/Core/src/AppHostBuilderExtensions.cs#L60

Should we move `[RUC]` to `UseMauiCompatibility()` instead of this interface?

I guess we could also consider a trimmer feature flag somehow?

Reply:
> I guess we could also consider a trimmer feature flag somehow?

Do you mean a feature flag for `Compatibility`?

One thing that confuses me that we previously marked the whole `Compatibility` assembly as `IsTrimmable` and I'm even changing that to `IsAotCompatible` in #21505. Honestly, maybe we should turn the analyzers for these assemblies off if we don't care about them? The alternative of marking the entry point as `[RUC]` and suppressing warnings elsewhere might be good enough, even though it could report warnings even though the developer uses just trimming-safe APIs from compatibility.

Reply:
Maybe we just file an issue to `[Obsolete]` `INativeBindingService` in the future.

Reply:
I opened the issue: https://github.com/dotnet/maui/issues/21861